src/test/java/org/pesmypetcare/webservice/service/MedicationServiceTest.java

```diff
@@ -0,0 +1,159 @@
+                isA(MedicationEntity.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenMedicationDeleted() {
+        service.deleteByDateAndName(owner, petName, dateName);
+        verify(medicationDao).deleteByDateAndName(isA(String.class), isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenAllMedicationsDeleted() throws DatabaseAccessException {
+        service.deleteAllMedications(owner, petName);
+        verify(medicationDao).deleteAllMedications(isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenAllMedicationDeleteFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(medicationDao).deleteAllMedications(any(String.class),
+                    isA(String.class));
+            service.deleteAllMedications(owner, petName);
+        }, "Should return a database access exception when a Medication deletion fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityWhenMedicationRetrieved() throws DatabaseAccessException {
+        when(medicationDao.getMedicationData(owner, petName, dateName)).thenReturn(medicationEntity);
+        MedicationEntity Medication = service.getMedicationData(owner, petName, dateName);
+        assertSame(medicationEntity, Medication, "Should return a Medication entity");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetMedicationRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(medicationDao).getMedicationData(any(String.class),
+                    any(String.class), isA(String.class));
+            service.getMedicationData(owner, petName, dateName);
+        }, GET_EXCEPTION_MSG);
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityListWhenGetSetOfMedicationsRetrieved() throws DatabaseAccessException,
+            ExecutionException, InterruptedException {
+        when(medicationDao.getAllMedicationData(owner, petName)).thenReturn(medicationList);
+        List<Map<List<String>, Object>> list = service.getAllMedicationData(owner, petName);
+        assertSame(medicationList, list, "Should return a list of Medication entities");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetSetOfMedicationsRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(medicationDao).getAllMedicationData(any(String.class),
+                    isA(String.class));
+            service.getAllMedicationData(owner, petName);
+        }, GET_EXCEPTION_MSG);
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityListWhenGetMedicationsBetweenDatesRetrieved() throws DatabaseAccessException,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 121).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)